core/src/test/java/org/elasticsearch/index/query/TermsQueryBuilderTest.java

```diff
@@ -0,0 +1,131 @@
+
+    @Override
+    protected TermsQueryBuilder doCreateTestQueryBuilder() {
+        TermsQueryBuilder query = newRandomQueryBuilder();
+
+        // NO COMMIT: untested terms lookup
+        return query;
+    }
+
+    private TermsQueryBuilder newRandomQueryBuilder() {
+        String fieldName = null;
+        List<Object> values = new ArrayList<>();
+        switch (randomIntBetween(0, 2)) {
+            case 0:
+                if (randomBoolean()) {
+                    fieldName = STRING_FIELD_NAME;
+                }
+                if (frequently()) {
+                    for (int i = 0; i < randomInt(5); i++) {
+                        values.add(randomAsciiOfLengthBetween(1, 10));
+                    }
+                } else {
+                    // generate unicode strings in 10% of cases
+                    for (int i = 0; i < randomInt(5); i++) {
+                        values.add(randomUnicodeOfLength(10));
+                    }
+                }
+                break;
+            case 1:
+                if (randomBoolean()) {
+                    fieldName = INT_FIELD_NAME;
+                }
+                for (int i = 0; i < randomInt(5); i++) {
+                    values.add(randomInt(10000));
+                }
+                break;
+            case 2:
+                if (randomBoolean()) {
+                    fieldName = DOUBLE_FIELD_NAME;
+                }
+                for (int i = 0; i < randomInt(5); i++) {
+                    values.add(randomDouble());
+                }
+                break;
+            default:
+                throw new UnsupportedOperationException();
+        }
+        if (fieldName == null) {
+            fieldName = randomAsciiOfLengthBetween(1, 10);
+        }
+        return new TermsQueryBuilder(fieldName, values);
+    }
+
+    @Override
+    protected void doAssertLuceneQuery(TermsQueryBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+        List<Object> terms = new ArrayList<>();
+        // NO COMMIT: untested we would need to mock a client in BaseQueryTestCase here
+        // we could also more simply mock handleTermsLookup
+        if (queryBuilder.isTermsLookupQuery()) {
+            TermsQueryBuilder.handleTermsLookup(queryBuilder.termsLookup().toTermsLookup(context), context.client(), terms);
+        } else {
+            terms.addAll(queryBuilder.values());
+        }
+        Query newQuery;
+        if (terms.isEmpty()) {
+            newQuery = Queries.newMatchNoDocsQuery();
+        } else {
+            newQuery = TermsQueryBuilder.handleTermsQuery(terms, queryBuilder.fieldName(), context,
+                    queryBuilder.minimumShouldMatch(), queryBuilder.disableCoord());
+        }
+        // NO COMMIT: the boost gets set when the query is rewritten, so we manually set it here
+        newQuery.setBoost(queryBuilder.boost());
+        assertEquals(query, newQuery);
```

Comment:
because of the way the constructors are setup we end with an empty list of values, but not with values set to null.